File: projects/Puerts_Demo/Assets/Examples/06_UIEvent/UIEvent.cs
using System;
using UnityEngine;
using Puerts;

public class UIEvent : MonoBehaviour
{
    static JsEnv jsEnv;

    void Start()
    {
        if (jsEnv == null)
        {
            jsEnv = new JsEnv();
            jsEnv.UsingAction<bool>();//toggle.onValueChanged用到
        }

        var init = jsEnv.Eval<Action<MonoBehaviour>>("const m = require('UIEvent'); m.init;");

        if (init != null) init(this);
    }
}

File: projects/Puerts_Demo/TsProj/UIEvent.ts
import {UnityEngine} from 'csharp'
import {$typeof} from 'puerts'

function init(monoBehaviour: UnityEngine.MonoBehaviour): void {
    let button  = monoBehaviour.GetComponent($typeof(UnityEngine.UI.Button)) as UnityEngine.UI.Button;
    let input = monoBehaviour.transform.parent.GetComponentInChildren($typeof(UnityEngine.UI.InputField)) as UnityEngine.UI.InputField;
    let toggle = monoBehaviour.transform.parent.GetComponentInChildren($typeof(UnityEngine.UI.Toggle)) as UnityEngine.UI.Toggle;
    button.onClick.AddListener(() =>{
        console.log("button pressed..., input is: " + input.text);
    });
    toggle.onValueChanged.AddListener((b) => {
        console.log("toggle.value=" + b);
    });
}

export {init};
